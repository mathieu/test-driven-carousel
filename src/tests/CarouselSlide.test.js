import React from 'react';
import { shallow } from 'enzyme';
import CarouselSlide from '../CarouselSlide';

describe('CarouselSlide', () => {
  let wrapper;
  const imgUrl = 'https://example.com/image.png';

  beforeEach(() => {
    wrapper = shallow(
      <CarouselSlide imgUrl={imgUrl} description="default description" />
    );
  });

  it('renders a <figure>', () => {
    expect(wrapper.type()).toBe('figure');
  });

  it('renders an <img> and a <figcaption> as children', () => {
    expect(wrapper.childAt(0).type()).toBe('img');
    expect(wrapper.childAt(1).type()).toBe('figcaption');
  });

  it('passes `imgUrl` through to the <img>', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(imgUrl);
  });

  it('uses `description` and attribution` as the figcaption', () => {
    const description = 'A jaw-droppinglt spectacular image';
    const attribution = 'Trevor Burnham';
    wrapper.setProps({ description, attribution });
    expect(wrapper.find('figcaption').text()).toBe(
      `${description} ${attribution}`
    );
    expect(wrapper.find('figcaption strong').text()).toBe(description);
  });

  it('passes extra props to the <figure>', () => {
    const className = 'myFigure';
    const style = {};
    const onClick = {};
    wrapper.setProps({ className, style, onClick });
    expect(wrapper.prop('className')).toBe(className);
    expect(wrapper.prop('style')).toBe(style);
    expect(wrapper.prop('onClick')).toBe(onClick);
  });
});
