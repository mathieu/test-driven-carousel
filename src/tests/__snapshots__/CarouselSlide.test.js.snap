// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselSlide renders correctly 1`] = `
<figure
  attribute="Attribute"
>
  <CarouselSlide__DefaultImg
    imgHeight={500}
    src="https://example.com/image.png"
  />
  <figcaption>
    <strong>
      Decription
    </strong>
     
  </figcaption>
</figure>
`;

exports[`Img renders correctly 1`] = `
.c0 {
  object-fit: cover;
  width: 100%;
  height: 500px;
}

<img
  className="c0"
  src="https://example.com/image.png"
/>
`;
